src/Driver/AbstractOracleDriver/EasyConnectString.php

```diff
@@ -55,15 +55,7 @@ public static function fromConnectionParameters(array $params): self
         $connectData = [];
 
         if (isset($params['servicename']) || isset($params['dbname'])) {
```

Comment:
Based on:
- https://github.com/gvenzl/oci-oracle-xe/issues/238#issuecomment-1920591949
- https://github.com/gvenzl/oci-oracle-free/issues/34

the service name should be the default, SID is internal DB identifier, but service name is more or less like "DB name", does support aliasing or clustering.